treelib/tree.py

```diff
@@ -801,23 +802,23 @@ def show(self, nid=None, level=ROOT, idhidden=True, filter=None,
         :param reverse: the ``reverse`` param for sorting :class:`Node` objects in the same level.
         :param line_type:
         :param data_property: the property on the node data object to be printed.
+        :param end: the string to append to the final line of the output. By
+            default, an extra newline is printed; pass '\n' if this is not
+            desired.
         :return: None
         """
-        self._reader = ""
+        self._reader = []
```

Comment:
IMO there is no reason to have a member variable as @GalacticStarfish explains it in https://github.com/caesar0301/treelib/issues/91. What do you think of replacing `self._reader` by a `reader` variable.

Reply:
+1, that's just a wast of memory cause it is cached but not reused AFAICS.